addons/product/models/product_attribute.py

```diff
@@ -447,7 +447,14 @@ def write(self, values):
                         _("You cannot change the product of the value %s set on product %s.") %
                         (ptav.display_name, ptav.product_tmpl_id.display_name)
                     )
-        return super(ProductTemplateAttributeValue, self).write(values)
+        res = super(ProductTemplateAttributeValue, self).write(values)
+
+        exclude_in_value = 'exclude_for' in values
+        if exclude_in_value:
+            self._unarchive_variants()
```

Comment:
This looks wrong, wouldn't it un-archive old variants that don't match any combination?
I don't understand why any variant should be un-archived in the process of wanting to archive even more of them.